nx/lib/nx/defn/compiler.ex

```diff
@@ -141,8 +141,16 @@ defmodule Nx.Defn.Compiler do
               if function_exported?(module, function, length(args)) do
                 formatted = Exception.format_mfa(module, function, length(args))
 
-                reraise "cannot invoke #{formatted} inside defn because it was not defined with defn",
-                        stack
+                message =
+                  "cannot invoke #{formatted} inside defn because it was not defined with defn"
+
+                detail =
+                  case module do
+                    IO -> ". Use inspect_value/2 or inspect_expr/2 instead"
```

Comment:
Let’s try to add a bit more information.  We can say,  if you want to print the runtime value of a tensor, use inspect_value. To print the tensor expression at compile time, use inspect_expr.

Reply:
Updated! Perhaps "compile time" is a bit confusing for someone knowing Elixir and new to Nx. Maybe we can word it differently, like "at compilation stage"?